compose.go

```diff
@@ -0,0 +1,213 @@
+package testcontainers
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const (
+	envProjectName = "COMPOSE_PROJECT_NAME"
+	envComposeFile = "COMPOSE_FILE"
+)
+
+// DockerCompose defines the contract for running Docker Compose
+type DockerCompose interface {
+	Down() ExecError
+	Invoke() ExecError
+	WithCommand([]string) DockerCompose
+	WithEnv(map[string]string) DockerCompose
+}
+
+// LocalDockerCompose represents a Docker Compose execution using local binary
+// docker-compose or docker-compose.exe, depending on the underlying platform
+type LocalDockerCompose struct {
+	Executable          string
+	ComposeFilePaths    []string
+	absComposeFilePaths []string
+	Identifier          string
+	Cmd                 []string
+	Env                 map[string]string
+}
+
+// NewLocalDockerCompose returns an instance of the local Docker Compose
+func NewLocalDockerCompose(filePaths []string, identifier string) *LocalDockerCompose {
+	dc := &LocalDockerCompose{}
+
+	dc.Executable = "docker-compose"
+	if runtime.GOOS == "windows" {
+		dc.Executable = "docker-compose.exe"
+	}
+
+	dc.ComposeFilePaths = filePaths
+
+	dc.absComposeFilePaths = make([]string, len(filePaths))
+	for i, cfp := range dc.ComposeFilePaths {
+		abs, _ := filepath.Abs(cfp)
+		dc.absComposeFilePaths[i] = abs
+	}
+
+	dc.Identifier = strings.ToLower(identifier)
+
+	return dc
+}
+
+// Down executes docker-compose down
+func (dc *LocalDockerCompose) Down() ExecError {
+	return executeCompose(dc, []string{"down"})
+}
+
+func (dc *LocalDockerCompose) getDockerComposeEnvironment() map[string]string {
+	environment := map[string]string{}
+
+	composeFileEnvVariableValue := ""
+	for _, abs := range dc.absComposeFilePaths {
+		composeFileEnvVariableValue += abs + string(os.PathListSeparator)
+	}
+
+	environment[envProjectName] = dc.Identifier
+	environment[envComposeFile] = composeFileEnvVariableValue
+
+	return environment
+}
+
+// Invoke invokes the docker compose
+func (dc *LocalDockerCompose) Invoke() ExecError {
+	return executeCompose(dc, dc.Cmd)
+}
+
+// WithCommand assigns the command
+func (dc *LocalDockerCompose) WithCommand(cmd []string) DockerCompose {
+	dc.Cmd = cmd
+	return dc
+}
+
+// WithEnv assigns the environment
+func (dc *LocalDockerCompose) WithEnv(env map[string]string) DockerCompose {
+	dc.Env = env
+	return dc
+}
+
+// ExecError is super struct that holds any information about an execution error, so the client code
+// can handle the result
+type ExecError struct {
```

Comment:
hey @mdelapenya sorry to add another suggestion so late 😬 but would it be a good idea to add the command (`cmd`) that failed to this struct?  I think that would be useful for debugging.  I know this is not trivial since the `Cmd` struct looks somewhat large.  Just and idea, wondering what you think?

Reply:
Sounds reasonable to me :wink:

Reply:
awesome thank you!

Reply:
Hey, I added info about the process: dirContext, binary name and arguments. I preferred not to add environment to avoid disclosing sensible information.

Please take a look :)

Reply:
looks good thank you! in the CI/CD build it looks like the cmd is blank (an empty slice)...could that be what is causing the builds to fail?

Reply:
Yeah, but not sure how the CMD was lost. I'll try to debug it locally and see what happens, because that test was passing from the beginning